test/t8_forest/t8_gtest_search.cxx

```diff
@@ -60,96 +61,81 @@ class forest_search: public testing::TestWithParam<std::tuple<t8_eclass, int>> {
-    EXPECT_TRUE (query != NULL) << "query " << iquery << " is NULL.";
+  EXPECT_FALSE (queries.empty ()) << "query callback must be called with queries argument. ";
+  EXPECT_EQ (active_query_indices.size (), (long unsigned int) 1)
+    << "Wrong number of active queries passed to query callback.";
+  for (int iquery : active_query_indices) {
     /* The query is an int with value 42 (see below) */
-    EXPECT_EQ (*(int *) query, 42) << "Wrong query argument passed to query callback.";
+    EXPECT_EQ (queries[iquery], 42) << "Wrong query argument passed to query callback.";
     if (is_leaf) {
       /* Test whether tree_leaf_index is actually the index of the element */
       t8_locidx_t test_ltreeid;
-      t8_eclass_t tree_class = t8_forest_get_tree_class (forest, ltreeid);
-      t8_eclass_scheme_c *ts;
-      ts = t8_forest_get_eclass_scheme (forest, tree_class);
+      const t8_eclass_t tree_class = t8_forest_get_tree_class (forest, ltreeid);
+      const t8_eclass_scheme_c *ts = t8_forest_get_eclass_scheme (forest, tree_class);
 
-      t8_locidx_t tree_offset = t8_forest_get_tree_element_offset (forest, ltreeid);
-      t8_element_t *test_element = t8_forest_get_element (forest, tree_offset + tree_leaf_index, &test_ltreeid);
+      const t8_locidx_t tree_offset = t8_forest_get_tree_element_offset (forest, ltreeid);
+      const t8_element_t *test_element = t8_forest_get_element (forest, tree_offset + tree_leaf_index, &test_ltreeid);
       EXPECT_ELEM_EQ (ts, element, test_element);
       EXPECT_EQ (ltreeid, test_ltreeid) << "Tree mismatch in search.";
     }
-    query_matches[iquery] = 1;
+    query_matches[iquery] = true;
   }
 }
 
 TEST_P (forest_search, test_search_one_query_matches_all)
 {
-  const int query = 42;
-  sc_array_t queries;
-  sc_array_t matched_leaves;
-
   /* set up a single query containing our query */
-  sc_array_init_size (&queries, sizeof (int), 1);
-  *(int *) sc_array_index (&queries, 0) = query;
+  std::vector<int> queries = { 42 };
 
   t8_locidx_t num_elements = t8_forest_get_local_num_elements (forest);
   /* set up an array in which we flag whether an element was matched in the
    * search */
-  sc_array_init_size (&matched_leaves, sizeof (int), num_elements);
-  /* write 0 in every entry */
-  for (t8_locidx_t ielement = 0; ielement < num_elements; ++ielement) {
-    *(int *) t8_sc_array_index_locidx (&matched_leaves, ielement) = 0;
-  }
+  std::vector<bool> matched_leaves (num_elements, false);
 
-  /* Set the array as user data so that we can access it in the search callback */
-  t8_forest_set_user_data (forest, &matched_leaves);
   /* Call search. This search matches all elements. After this call we expect
    * all entries in the matched_leaves array to be set to 1. */
 
-  t8_forest_search (forest, t8_test_search_all_fn, t8_test_search_query_all_fn, &queries);
+  t8_search_with_queries<int, std::vector<bool>> search (t8_test_search_all_fn, t8_test_search_query_all_fn, queries);
```

Comment:
we should also test the t8_search class (without queries).

Reply:
Just added a new test for this.